### ISSUE TITLE
Question about scopes in different containers

### ISSUE THREAD

**Issue by IvanShah** [hmlongco/Factory#86]
Hello I have a problem with scopes of some services which I want share between containers as singletons. The service created as many times as it injects by different containers. 
In the next example sharedService inits twice when MyConstructedService is created and when sharedService injected by something else. What I need: only one sharedService ever. How I can do that?

```
extension SharedContainer {
    var sharedService: Factory<MyServiceType> { self { MyService() }.singleton }
}

final class DemoContainer: ObservableObject, SharedContainer {
    static var shared = DemoContainer()

    var constructedService: Factory<MyConstructedService> {
        self {
            MyConstructedService(service: sharedService())
        }
    }

    var manager = ContainerManager()
}
```

Thank you

**Comment by hmlongco** [comment]
I think singletons may need so rework as the way they're implemented in 2.0 gives rise to the issues you've illustrated.

Given the current architecture, you may do do something like...
```swift
final class SingletonContainer: SharedContainer {
    static var shared = SingletonContainer()
    var singletonService: Factory<MyServiceType> {
        self { MyService() }.singleton
    }
    var manager = ContainerManager()
}

extension SharedContainer {
    var sharedService: Factory<MyServiceType> {
        self { SingletonContainer.shared.singletonService() }
    }
}

final class DemoContainer: SharedContainer {
    static var shared = DemoContainer()
    var constructedService: Factory<MyConstructedService> {
        self { MyConstructedService(service: self.sharedService()) }
    }
    var manager = ContainerManager()
}
```
It's clunky, but the best I can think of until I get a workaround implemented.

**Comment by IvanShah** [comment]
Thank you for workaround. Looking forward to update
